pallets/capacity/src/lib.rs

```diff
@@ -582,19 +647,31 @@ impl<T: Config> Pallet<T> {
 	) -> Result<BalanceOf<T>, DispatchError> {
 		let mut staking_account =
 			Self::get_staking_account_for(unstaker).ok_or(Error::<T>::StakingAccountNotFound)?;
-		ensure!(amount <= staking_account.active, Error::<T>::AmountToUnstakeExceedsAmountStaked);
+		ensure!(amount <= staking_account.active, Error::<T>::InsufficientStakingBalance);
 
-		let current_epoch: T::EpochNumber = Self::get_current_epoch();
-		let thaw_period = T::UnstakingThawPeriod::get();
-		let thaw_at = current_epoch.saturating_add(thaw_period.into());
-
-		let unstake_result = staking_account.withdraw(amount, thaw_at)?;
+		let unstake_result = staking_account.withdraw(amount, Self::get_thaw_at_epoch())?;
 
 		Self::set_staking_account(&unstaker, &staking_account);
 
 		Ok(unstake_result)
 	}
 
+	fn get_thaw_at_epoch() -> <T as Config>::EpochNumber {
+		let current_epoch: T::EpochNumber = Self::get_current_epoch();
+		let thaw_period = T::UnstakingThawPeriod::get();
+		current_epoch.saturating_add(thaw_period.into())
+	}
+
+	fn ensure_can_unstake(unstaker: &T::AccountId) -> Result<(), DispatchError> {
+		let staking_account: StakingAccountDetails<T> =
```

Comment:
I had to pull some logic out of reduce_capacity so that change_staking_target could use it, however it was checking for unlock and staking change chunks which made some things fail that didn't need to.  So `ensure_can_unstake` was created, and is now called directly from the `unstake` extrinsic.